### PR TITLE
Simplify case search case property queries

### PR DESCRIPTION
**Pull request by esoergel**
## Technical Summary
This is a few small clean-ups to case search query logic.

### Safety story

This code is quite well tested, and the changes are logically equivalent

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/case_search/tests/test_filter_dsl.py [hunk 1]
```diff
@@ -488,46 +479,30 @@ def test_selected(self):
     def test_selected_any(self):
         parsed = parse_xpath("selected-any(first_name, 'Jon John Jhon')")
         expected_filter = {
-            "bool": {
```
Comment (orangejenny) [inline corehq/apps/case_search/tests/test_filter_dsl.py:491]: Every time you remove a level of indentation, an angel gets its wings.

**Review by orangejenny — APPROVED** [review]

**Review by joxl — APPROVED** [review]
LGTM from the perspective of the Elastic work I've been up to lately (which admittedly hadn't gotten into `ESQuery` much).

File: corehq/apps/es/case_search.py [hunk 2]
```diff
@@ -285,10 +285,7 @@ def case_property_missing(case_property_name):
         filters.NOT(
             queries.nested(
                 CASE_PROPERTIES_PATH,
-                queries.filtered(
-                    queries.match_all(),
-                    filters.term(PROPERTY_KEY, case_property_name),
-                )
+                filters.term(PROPERTY_KEY, case_property_name),
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/es/case_search.py:288]: do you know why this was added in the first place?

[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/es/case_search.py:288]: Based on ES docs it seems like having the match_all would give every nested doc a score of `1` which doesn't seem like something we'd want.

**Review by snopoke — APPROVED** [review]

File: corehq/apps/es/case_search.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/es/case_search.py:288]: I assumed it was added due to the muddiness between filters and queries - like, to use a filter in a query context, do you need a "filtered query"?  It does bug me considerably how much we use queries when we really only want un-scored filtered results.  It's sometimes hard to say for sure with case search, though.

**Pull request merged**